File: app/src/main/java/com/example/myapplication/data/network/ProDetailsNetwork.java
package com.example.myapplication.data.network;

import com.example.myapplication.data.bean.ProDetails;
import com.example.myapplication.data.network.Service.ProDetailsService;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ProDetailsNetwork {
    ProDetails proDetails;

    public ProDetails fetchProDetails(String category) {
        ProDetailsService service = ServiceCreator.create("https://ip地址/merchandiseList/",
                ProDetailsService.class);
        Call<ProDetails> call = service.getProDetails(category);
        call.enqueue(new Callback<ProDetails>() {
            @Override
            public void onResponse(Call<ProDetails> call, Response<ProDetails> response) {
                if (response.isSuccessful()) {
                    proDetails = response.body();
                }
            }

            @Override
            public void onFailure(Call<ProDetails> call, Throwable t) {

            }
        });
        return proDetails;
    }
}

File: app/src/main/java/com/example/myapplication/data/network/ProDetailsNetwork.kt
package com.example.myapplication.data.network

import com.example.myapplication.data.bean.ProDetailsInfo
import com.example.myapplication.data.network.Service.ProDetailsService
import com.example.myapplication.utils.await

class ProDetailsNetwork {
    private val proDetailsService = ServiceCreator
            .create("http://114.55.145.30:8090/school/second/", ProDetailsService::class.java)

    suspend fun fetchProDetails(id: Int): ProDetailsInfo  = proDetailsService.getProDetails(id).await().detailsInfo

    companion object {
        private var network: ProDetailsNetwork? = null
        fun getInstance(): ProDetailsNetwork? {
            if (network == null) {
                synchronized(ProDetailsNetwork::class.java) {
                    if (network == null) {
                        network = ProDetailsNetwork()
                    }
                }
            }
            return network
        }
    }
}